modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt

```diff
@@ -243,32 +242,50 @@ open class Iroha2Client(
         }
     }
 
+    fun subscribeToBlockStream(from: Long = 1, count: Int) = subscribeToBlockStream(
+        from,
+        count,
+        action = { block -> block },
+    )
+
     /**
      * Subscribe to block streaming
      * @param from - block number to start from
-     * @param count - how many blocks to get before closing web socket
+     * @param count - how many blocks to get before closing the channel
+     * @param action - code which will be invoked after a new block received
+     * @param closeOn - if the condition returns true then the channel will be closed
      */
-    fun subscribeToBlockStream(from: Long, count: Int): Flow<VersionedBlockMessage> = flow {
+    fun <T> subscribeToBlockStream(
+        from: Long = 1,
+        count: Int? = null,
+        action: suspend (block: VersionedBlockMessage) -> T,
+        closeOn: suspend (block: VersionedBlockMessage) -> Boolean = { false },
```

Comment:
```suggestion
        closeOnBlockHeight: suspend (block: VersionedBlockMessage) -> Boolean = { false },
```

Reply:
the logic of `closeOn` can be tied not to `height`

Reply:
`closeOnBlockCondition` or smth like that then?

Reply:
I'd leave like it is

Reply:
or `closeIf`

Reply:
yes, that's better than nothing, `closeOn` may confuse referring to `onClose` (one time action when closing)